README.md

```diff
@@ -29,122 +29,122 @@ Related projects:
+You can enable **caching** to support those high-volume requests.  
 
-NOTES:
-* The read after write will become eventual consistent, unless set bypassCachingForStrongConsistency=true in RPC options
-* Caching will be more useful for read-only RPC(no persistence.SetXXX API or communication API calls in RPC implementation) or GetDataAttributes API.
-  * A read-only RPC can still invoke any other RPCs(like calling other microservices, or DB operation) in the RPC implementation
-* It will cost extra event in history (upsertMemo operation for WorkflowPropertiesModified event) for updating the persisted data attributes
-* This feature is currently only supported if the backend is Temporal, because [Cadence doesn't support mutable memo](https://github.com/uber/cadence/issues/3729)
+Note:
+* With caching enabled read-after-write access will become *eventually consistent*, unless `bypassCachingForStrongConsistency=true` is set in RPC options
+* Caching will introduce an extra event in history (upsertMemo operation for WorkflowPropertiesModified event) for updating the persisted data attributes
+* Caching will be more useful for read-only RPC (no persistence.SetXXX API or communication API calls in RPC implementation) or GetDataAttributes API.
+  * A read-only RPC can still invoke any other RPCs (like calling other microservices, or DB operation) in the RPC implementation
+* Caching is currently only supported if the backend is Temporal, because [Cadence doesn't support mutable memo](https://github.com/uber/cadence/issues/3729)
 
 ## Workflow State
-WorkflowState is to implement the asynchronous process as "workflow".  
-It will be running in the background, with default infinite backoff retry. 
+WorkflowState is how you implement your asynchronous process as a "workflow".  
+It will run in the background, with infinite backoff retry by default. 
  
-A WorkflowState is like “a small workflow” of 1~2 steps:
+A WorkflowState is itself like “a small workflow” of 1 or 2 steps:
 
-**[ waitUntil ] → execute**
+**[ `waitUntil` ] → `execute`**
 
-The `waitUntil` API can returns some commands to wait for. When the commands are completed, the `execute` API will be invoked.
-The two APIs have access to read/write the persistence defined in the workflow.
+The `waitUntil` API returns "commands" to wait for. When the commands are completed, the `execute` API will be invoked.
+Both the `waitUntil` and `execute` APIs have access to read/write the persistence schema defined in the workflow.
 
-The full detailed execution flow is like this:
+The full execution flow looks like this:
 
 ![Workflow State diagram](https://user-images.githubusercontent.com/4523955/234921554-587d8ad4-84f5-4987-b838-959869293465.png)
 
-The `waitUntil` API is optional. If not defined, then `execute` API will be invoked instead when the state started.
+The `waitUntil` API is optional. If not defined, then the `execute` API will be invoked immediately when the Workflow State is started.
```

Comment:
I've changed this to Workflow State. But I don't know if that's clearer or less clear. Internally I guess they are called just states. I could go with capitalized State here if that's more correct.

Reply:
Definitely better and clearer! Sometimes it’s called “state” just for short, esp in verbal communication . But it’s better to use the full name here.